lib/internal/fs/utils.js

```diff
@@ -462,15 +463,47 @@ function BigIntStats(dev, mode, nlink, uid, gid, rdev, blksize,
   this.mtimeNs = mtimeNs;
   this.ctimeNs = ctimeNs;
   this.birthtimeNs = birthtimeNs;
-  this.atime = dateFromMs(this.atimeMs);
-  this.mtime = dateFromMs(this.mtimeMs);
-  this.ctime = dateFromMs(this.ctimeMs);
-  this.birthtime = dateFromMs(this.birthtimeMs);
 }
 
 ObjectSetPrototypeOf(BigIntStats.prototype, StatsBase.prototype);
 ObjectSetPrototypeOf(BigIntStats, StatsBase);
 
+ObjectDefineProperty(BigIntStats.prototype, 'atime', {
+  enumerable: true,
+  get: function() {
+    const value = dateFromMs(this.atimeMs);
+    ObjectDefineProperty(this, 'atime', { __proto__: null, value });
```

Comment:
There's no way to use it without `ObjectDefineProperty`? I think it will make getting this data much slower.

Reply:
The initial operation will work as a getter and later, with the help of ObjdctDefineProperty we convert it into a property due avoid recreating a date. I don't think there is another way of doing this.

Reply:
If we can route some of this logic into cpp the `SetLazyDataProperty` helper is probably the most efficient way of doing this.  Probably not if we're bouncing in and out of cpp land, but if the actual date construction can also be done in cpp then that's probably the route you want to take